tests/test_l1dbproto.py

```diff
@@ -0,0 +1,475 @@
+import lsst.afw.image.utils as afwImageUtils
+import lsst.afw.geom as afwGeom
+import lsst.afw.table as afwTable
+import lsst.daf.base as dafBase
+import lsst.pipe.base as pipeBase
+import lsst.utils.tests
+
+
+def create_test_points(point_locs_deg,
+                       start_id=0,
+                       schema=None,
+                       scatter_arcsec=1.0,
+                       indexer_ids=None,
+                       associated_ids=None):
+    """Create dummy DIASources or DIAObjects for use in our tests.
+
+    Parameters
+    ----------
+    point_locs_deg : array-like (N, 2) of `float`s
+        Positions of the test points to create in RA, DEC.
+    start_id : `int`
+        Unique id of the first object to create. The remaining sources are
+        incremented by one from the first id.
+    schema : `lsst.afw.table.Schema`
+        Schema of the objects to create. Defaults to the DIASource schema.
+    scatter_arcsec : `float`
+        Scatter to add to the position of each DIASource.
+    indexer_ids : `list` of `ints`s
+        Id numbers of pixelization indexer to store. Must be the same length
+        as the first dimension of point_locs_deg.
+    associated_ids : `list` of `ints`s
+        Id numbers of associated DIAObjects to store. Must be the same length
+        as the first dimension of point_locs_deg.
+
+    Returns
+    -------
+    test_points : `lsst.afw.table.SourceCatalog`
+        Catalog of points to test.
+    """
+    if schema is None:
+        schema = make_minimal_dia_source_schema()
+    sources = afwTable.SourceCatalog(schema)
+
+    for src_idx, (ra, dec,) in enumerate(point_locs_deg):
+        src = sources.addNew()
+        src['id'] = src_idx + start_id
+        coord = afwGeom.SpherePoint(ra, dec, afwGeom.degrees)
+        if scatter_arcsec > 0.0:
+            coord = coord.offset(
+                np.random.rand() * 360 * afwGeom.degrees,
+                np.random.rand() * scatter_arcsec * afwGeom.arcseconds)
+        if indexer_ids is not None:
+            src['pixelId'] = indexer_ids[src_idx]
+        if associated_ids is not None:
+            src['diaObjectId'] = associated_ids[src_idx]
+        src.setCoord(coord)
+
+    return sources
+
+
+class TestAssociationDBSqlite(unittest.TestCase):
+
+    def setUp(self):
+        """Initialize an empty database.
+        """
+
+        # CFHT Filters from the camera mapper.
+        afwImageUtils.resetFilters()
+        afwImageUtils.defineFilter('u', lambdaEff=374, alias="u.MP9301")
+        afwImageUtils.defineFilter('g', lambdaEff=487, alias="g.MP9401")
+        afwImageUtils.defineFilter('r', lambdaEff=628, alias="r.MP9601")
+        afwImageUtils.defineFilter('i', lambdaEff=778, alias="i.MP9701")
+        afwImageUtils.defineFilter('z', lambdaEff=1170, alias="z.MP9801")
+        afwImageUtils.defineFilter('y', lambdaEff=2000, alias="y.MP9901")
+
+        (self.tmp_file, self.db_file) = tempfile.mkstemp(dir=os.path.dirname(__file__))
```

Comment:
I can't figure out why you need the parens around self.temp_file and self.db_file.